src/uct/sm/mm/posix/mm_posix.c

```diff
@@ -419,11 +419,21 @@ uct_posix_mem_alloc(uct_md_h tl_md, size_t *length_p, void **address_p,
                                                          uct_posix_md_config_t);
     ucs_status_t status;
     uct_mm_seg_t *seg;
+    uct_md_attr_t md_attr;
     int force_hugetlb;
     int mmap_flags;
     void *address;
     int fd;
 
+    status = uct_posix_md_query(tl_md, &md_attr);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    if (!(md_attr.cap.alloc_mem_types & UCS_BIT(mem_type))) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
```

Comment:
no need to query. posix supports only HOST

Reply:
done